### PR TITLE
[misc] Use dune build info for refined version view.

### PR DESCRIPTION
**Pull request by ejgallego**
For example on pins.

### CODE REVIEW

**Comment by Alizter** [comment]
@ejgallego Nix now works.

**Review by Alizter — APPROVED** [review]

File: controller/lsp_core.ml [hunk 1]
```diff
@@ -274,7 +274,14 @@ let log_workspace w =
   LIO.trace "workspace" "initialized" ~extra;
   LIO.logMessage ~lvl:3 ~message
 
-let version () = Version.server
+let version () =
+  let dev_version =
+    match Build_info.V1.version () with
+    | None -> "NA"
```
Comment (Alizter) [inline controller/lsp_core.ml:280]:
```suggestion
    | None -> "N/A"
```
This is a de facto standard meaning "not applicable" or "not available".

File: flake.nix [hunk 2]
```diff
@@ -49,7 +49,7 @@
           in
             l.attrValues {
               inherit serapi;
-              inherit (ocamlPackages) yojson cmdliner uri;
+              inherit (ocamlPackages) yojson cmdliner uri dune-build-info;
```
Comment (Alizter) [inline flake.nix:52]: I should note that this is really `dune-build-info.2.6.0`. I am pretty sure that it hasn't changed in 3.6 and we should be safe under the `V1` API anyway. It doesn't appear to cause any ill-effects in the Nix builds.

**Comment by ejgallego** [comment]
Thanks a lot @Alizter , tweak to the message applied!

**Pull request merged**